Add HTML category to UnifiedWebPreferences.yaml

nt1m opened the pull request:
#### 139404b6f591d82bd665e03139871faa6f0ff5a9
<pre>
Add HTML category to UnifiedWebPreferences.yaml
<a href="https://bugs.webkit.org/show_bug.cgi?id=254352">https://bugs.webkit.org/show_bug.cgi?id=254352</a>
rdar://107144335

Reviewed by Brent Fulgham.

Lighten the DOM category by introducing a new HTML one, containing:
- inert attribute
- popover attribute
- Vertical form controls
- HTML &lt;model&gt; element
- HTML &lt;model&gt; document
- lazy image loading (&lt;img loading=lazy&gt;)
- lazy iframe loading (&lt;iframe loading=lazy&gt;)
- dialog element
- PDF.js (functionality wise similar to HTML &lt;model&gt; document, but also this is behind a build flag off by default, so doesn&apos;t matter where it goes as long as it can be turned on)

* Source/WTF/Scripts/GeneratePreferences.rb:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/UIProcess/API/APIFeatureStatus.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKFeature.mm:
(-[_WKFeature category]):
* Source/WebKitLegacy/mac/WebView/WebFeature.h:

Canonical link: <a href="https://commits.webkit.org/262100@main">https://commits.webkit.org/262100@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7048127203a819ecc95175484b3e6fee885551db

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/552 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/570 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/596 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/646 "Built successfully") | [❌ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/483 "Failed to checkout and rebase branch from PR 11937") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/618 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/656 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/725 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/561 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/531 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/528 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/651 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/592 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/514 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/555 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/489 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/490 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/535 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/553 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/522 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/534 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/575 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/477 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/530 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/101 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/131 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/531 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/574 "Built successfully") | 
| | | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/98 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -2993,8 +2993,8 @@ IndexedDBAPIEnabled:
-  category: css
-  humanReadableName: "inert attribute"
+  category: html
+  humanReadableName: "HTML inert attribute"
```
Do we need to prefix the name with HTML if it's in the new HTML category?

brentfulgham replied:
And if so, why don't you change the other labels, too?

nt1m replied:
Wanted to make it consistent with the popover attribute. Regarding whether to put the prefix or not, the same question applies to the CSS labels as well.

Personally I like the prefix, because it keeps the label capitalized without having to capitalize a technical term.

brentfulgham reviewed (approved):
Looks great!

patrickangle commented on Source/WebKitLegacy/mac/WebView/WebFeature.h:
```diff
@@ -59,6 +59,7 @@ typedef NS_ENUM(NSUInteger, WebFeatureCategory) {
     WebFeatureCategoryAnimation,
     WebFeatureCategoryCSS,
     WebFeatureCategoryDOM,
+    WebFeatureCategoryHTML,
```
I need to forward-declare this value, and likely others in the future if we add them. To make that possible, can we add the new value to the end and provide an explicit value for each of these?

nt1m committed:
Add HTML category to UnifiedWebPreferences.yaml
https://bugs.webkit.org/show_bug.cgi?id=254352
rdar://107144335

Reviewed by Brent Fulgham.

Lighten the DOM category by introducing a new HTML one, containing:
- inert attribute
- popover attribute
- Vertical form controls
- HTML <model> element
- HTML <model> document
- lazy image loading (<img loading=lazy>)
- lazy iframe loading (<iframe loading=lazy>)
- dialog element
- PDF.js (functionality wise similar to HTML <model> document, but also this is behind a build flag off by default, so doesn't matter where it goes as long as it can be turned on)

* Source/WTF/Scripts/GeneratePreferences.rb:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/UIProcess/API/APIFeatureStatus.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKFeature.mm:
(-[_WKFeature category]):
* Source/WebKitLegacy/mac/WebView/WebFeature.h:

Canonical link: https://commits.webkit.org/262100@main
